pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -594,9 +595,9 @@ private void updateMessageMetadata(final MessageMetadata msgMetadata, final int
         }
     }
 
-    private long updateMessageMetadataSequenceId(final MessageMetadata msgMetadata) {
+    private long updateMessageMetadataSequenceId(final MessageMetadata msgMetadata, boolean isChunk) {
         final long sequenceId;
-        if (!msgMetadata.hasSequenceId()) {
+        if (!msgMetadata.hasSequenceId() || isChunk) {
```

Comment:
It seems like we are overriding cx provided sequence id here, if chunked. Are we allowed to do this?

Reply:
Yes, this seems would break the behavior. We need to take care of it.

Reply:
Yeah, I will send an email to discuss it.

Reply:
~~Yeah, I will send an email to discuss it.~~
I optimized the logic. And now, it should not be a breaking change, because its behavior for users wouldn’t change.